MonoGame.Framework/CurveKeyCollection.cs

```diff
@@ -178,6 +179,25 @@ public int IndexOf(CurveKey item)
             return _keys.IndexOf(item);
         }
 
+        /// <summary>
+        /// Searches for the key with the lowest position greater than or equal to the specified position.
+        /// </summary>
+        /// <param name="position">Position to search for.</param>
+        /// <returns>The lowest index of the matching keys or, if the last element's position is less than the specified position, <c>Count</c>. </returns>
+        public int IndexAtPosition(float position)
+        {
+            int index = _keys.BinarySearch(new CurveKey(position, 0), _keyPositionComparer);
+            
+            //If position doesn't exist in list, return the key with the closest larger position
+            if (index < 0)
+                return ~index;
```

Comment:
Why not preserve the info for the developer? (result >=0 for an exact match or the negative of the closest match).
This check can be moved in `GetCurvePosition(float position)`